app/(api)/_datalib/collections/deleteCollectionItem.ts

```diff
@@ -0,0 +1,33 @@
+import { getDatabase } from '@utils/mongodb/mongoClient.mjs';
+import { NotFoundError } from '@utils/response/Errors';
+
+export async function deleteCollectionItem(collection: string, id: string) {
+  try {
+    const db = await getDatabase();
+    const reqCollection = await db.collection(collection);
+
+    const reqDocument = await reqCollection.findOneAndDelete({
+      id: id,
+    });
+
+    if (!reqDocument || reqDocument.length === 0) {
+      throw new NotFoundError(`No Items Found.`);
+    }
+
+    return { ok: true, body: 'CollectionItem deleted.', error: null };
+  } catch (error) {
+    if (error instanceof NotFoundError) {
```

Comment:
I would recommend doing:

```js
... catch(e) {
    const error = e as HttpError;
    return {
        ok: false,
        body: null,
        error: error.message || 'Internal Server Error',
    }
```

Also, note how in the createCollectionItem, we return a NextResponse object. Here, we are returning JSON? We can't have inconsistencies this big. Choose one or the other, whichever makes most sense to you. I think for HackDavis Judging app we returned NextResponses, but returning JSON is valid as well, if not more valid.

Choose one and stick to it. Maybe see what the developer experience is when using the datalib function in the api setting and the server actions setting to see which one you like better.